spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -0,0 +1,381 @@
+
+        case IS_NULL:
+          return isNull(toColumnName(child(predicate)));
+
+        case NOT_NULL:
+          return notNull(toColumnName(child(predicate)));
+
+        case LT:
+          if (rightChild(predicate) instanceof Literal) {
+            return lessThan(
+                toColumnName(leftChild(predicate)),
+                convertUTF8StringIfNecessary(rightChild(predicate)));
+          } else {
+            return greaterThan(
+                toColumnName(rightChild(predicate)),
+                convertUTF8StringIfNecessary(leftChild(predicate)));
+          }
+
+        case LT_EQ:
+          if (rightChild(predicate) instanceof Literal) {
+            return lessThanOrEqual(
+                toColumnName(leftChild(predicate)),
+                convertUTF8StringIfNecessary(rightChild(predicate)));
+          } else {
+            return greaterThanOrEqual(
+                toColumnName(rightChild(predicate)),
+                convertUTF8StringIfNecessary(leftChild(predicate)));
+          }
+
+        case GT:
+          if (rightChild(predicate) instanceof Literal) {
+            return greaterThan(
+                toColumnName(leftChild(predicate)),
+                convertUTF8StringIfNecessary(rightChild(predicate)));
+          } else {
+            return lessThan(
+                toColumnName(rightChild(predicate)),
+                convertUTF8StringIfNecessary(leftChild(predicate)));
+          }
+
+        case GT_EQ:
+          if (rightChild(predicate) instanceof Literal) {
+            return greaterThanOrEqual(
+                toColumnName(leftChild(predicate)),
+                convertUTF8StringIfNecessary(rightChild(predicate)));
+          } else {
+            return lessThanOrEqual(
+                toColumnName(rightChild(predicate)),
+                convertUTF8StringIfNecessary(leftChild(predicate)));
+          }
+
+        case EQ: // used for both eq and null-safe-eq
+          Object value;
+          String attributeName;
+          if (rightChild(predicate) instanceof Literal) {
+            attributeName = toColumnName(leftChild(predicate));
+            value = convertUTF8StringIfNecessary(rightChild(predicate));
+          } else {
+            attributeName = toColumnName(rightChild(predicate));
+            value = convertUTF8StringIfNecessary(leftChild(predicate));
+          }
+
+          if (predicate.name().equals("=")) {
+            // comparison with null in normal equality is always null. this is probably a mistake.
+            Preconditions.checkNotNull(
+                value, "Expression is always false (eq is not null-safe): %s", predicate);
+            return handleEqual(attributeName, value);
+          } else { // "<=>"
+            if (value == null) {
+              return isNull(attributeName);
+            } else {
+              return handleEqual(attributeName, value);
+            }
+          }
+
+        case IN:
+          return in(
```

Comment:
To have this as one pass, we could have:

```
if (isSupportedInPredicate(predicate)) {
  return in(
      toColumnName((NamedReference) predicate.children()[0]),
      Arrays.stream(predicate.children())
          .skip(1)
          .map(child -> convertLiteral(((Literal<?>) child)))
          .filter(Objects::nonNull)
          .collect(Collectors.toList()));
} else {
  return null;
}
```

Reply:
Changed. Thanks!